conference.js

```diff
@@ -1487,49 +1492,52 @@ export default {
         const didHaveVideo = Boolean(this.localVideo);
         const wasVideoMuted = this.isLocalVideoMuted();
 
-        return createLocalTracksF({
-            desktopSharingSourceDevice: options.desktopSharingSources
-                ? null : config._desktopSharingSourceDevice,
-            desktopSharingSources: options.desktopSharingSources,
-            devices: [ 'desktop' ],
-            desktopSharingExtensionExternalInstallation: {
-                interval: 500,
-                checkAgain: () => DSExternalInstallationInProgress,
-                listener: (status, url) => {
-                    switch (status) {
-                    case 'waitingForExtension': {
-                        DSExternalInstallationInProgress = true;
-                        externalInstallation = true;
-                        const listener = () => {
-                            // Wait a little bit more just to be sure that we
-                            // won't miss the extension installation
-                            setTimeout(
-                                () => {
+        const getDesktopStreamPromise = options.desktopStream
```

Comment:
It's hard to read the diff: is it the only change here to add `options.desktopStream` ?

Reply:
Yeah, that's the main change. If that desktopStream is already defined, skip doing createLocalTracksF and go down to adding the LOCAL_TRACK_STOPPED handler.